java-checks/src/main/java/org/sonar/java/checks/tests/SpringAssertionsSimplificationCheck.java

```diff
@@ -0,0 +1,109 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.tests;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.java.model.ExpressionUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import static org.sonar.plugins.java.api.semantic.MethodMatchers.ANY;
+
+@Rule(key = "S5970")
+public class SpringAssertionsSimplificationCheck extends IssuableSubscriptionVisitor {
+
+  private static final String MESSAGE_TEMPLATE = "Replace this assertion by \"%s\".";
+
+  private static final String[] ASSERTION_CLASSES = {
+    // JUnit4
+    "org.junit.Assert",
+    "junit.framework.TestCase",
+    // JUnit4 (deprecated)
+    "junit.framework.Assert",
+    // JUnit5
+    "org.junit.jupiter.api.Assertions"
+  };
+
+  public static final MethodMatchers ASSERT_EQUALS_MATCHER = MethodMatchers.create()
+    .ofTypes(ASSERTION_CLASSES)
+    .names("assertEquals")
+    .addParametersMatcher(ANY, ANY)
+    .build();
+
+  public static final MethodMatchers ASSERT_TRUE_FALSE_EQUALS_MATCHER = MethodMatchers.create()
+    .ofTypes(ASSERTION_CLASSES)
+    .names("assertEquals", "assertTrue", "assertFalse")
+    .withAnyParameters()
+    .build();
+
+  public static final MethodMatchers MODEL_VIEW_GET_VIEW_NAME = MethodMatchers.create()
+    .ofTypes("org.springframework.web.servlet.ModelAndView")
+    .names("getViewName")
+    .addWithoutParametersMatcher()
+    .build();
+
+  public static final MethodMatchers MODEL_MAP_GET = MethodMatchers.create()
+    .ofTypes("org.springframework.ui.ModelMap")
+    .names("get")
+    .addParametersMatcher("java.lang.Object")
+    .build();
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    MethodInvocationTree mit = (MethodInvocationTree) tree;
+    if (MODEL_VIEW_GET_VIEW_NAME.matches(mit)) {
+      getNestingCall(tree, ASSERT_EQUALS_MATCHER).ifPresent(id ->
+        reportIssue(id, String.format(MESSAGE_TEMPLATE, "ModelAndViewAssert.assertViewName"))
+      );
+    } else if (MODEL_MAP_GET.matches(mit)) {
+      getNestingCall(tree, ASSERT_TRUE_FALSE_EQUALS_MATCHER).ifPresent(id ->
+        reportIssue(id, String.format(MESSAGE_TEMPLATE, "ModelAndViewAssert.assertModelAttributeValue"))
+      );
+    }
+  }
+
+  private static Optional<IdentifierTree> getNestingCall(Tree nestedTree, MethodMatchers enclosingCall) {
+    Tree parent = nestedTree.parent();
+    while (parent != null && !parent.is(Tree.Kind.METHOD)) {
```

Comment:
As discussed, I agree that this pattern should be used with care. It is acceptable in this case though, since calls to Spring methods will hardly ever occur compared to common assertions.